### PR TITLE
Add option to change attachments on desktop build

### PR DESCRIPTION
**Pull request by suricactus**
Clicking on the gallery button opens a standard qt file dialog, instead of silently ignoring it.

### CODE REVIEW

**Comment by suricactus** [comment]
@m-kuhn let's merge this, we are waiting for the next opportunity in the numbers :)

**Comment by m-kuhn** [comment]
I agree there should be a desktop implementation for this, but I'm not convinced that the picture source is the right place for it. This handling is specific for the "gallery picture", whereas a different implementation could be envisioned for camera.
I think this logic/dialog should instead be in the `getGalleryPicture` implementation and no requirement should be there to call `init()` from QML.

**Comment by suricactus** [comment]
I did not find a way to return a `PictureSource` and in the mean time avoid calling `init`, because the rest of the implementation depends on the `pictureReceived` signal. If I emit it from the constructor, the listeners does not get triggered, as they are attached at later stage.

I do agree that `getGalleryPicture` is the place to go, I will try to find a solution

File: src/core/picturesource.h [hunk 1]
```diff
@@ -35,6 +35,9 @@ class PictureSource : public QObject
      * Emit this signal when a picture really has been received.
      */
     void pictureReceived( const QString &path );
+  protected:
```
Comment (m-kuhn) [inline src/core/picturesource.h:38]:
```suggestion
  private:
```

File: src/core/picturesource.cpp [hunk 2]
```diff
@@ -15,10 +15,19 @@
...
+
+PictureSource::PictureSource( QObject *parent, const QString &prefix, const QString &pictureFilePath )
   : QObject( parent )
+  , mPrefix( prefix )
+  , mPictureFilePath( pictureFilePath )
 {
+  if ( mPictureFilePath.startsWith( mPrefix ) )
+    mPictureFilePath = mPictureFilePath.remove( mPrefix );
 
+  QTimer::singleShot(0, this, [ = ] () {
```
Comment (m-kuhn) [inline src/core/picturesource.cpp:28]: Nice solution

File: src/core/picturesource.h (hunk 1 above)
Comment (suricactus) [reply src/core/picturesource.h]: done!

File: src/core/picturesource.h [hunk 3]
```diff
@@ -26,7 +26,7 @@ class PictureSource : public QObject
 {
     Q_OBJECT
   public:
-    explicit PictureSource( QObject *parent = nullptr );
+    explicit PictureSource( QObject *parent = nullptr, const QString &prefix = QString(), const QString &pictureFilePath = QString() );
```
Comment (m-kuhn) [inline src/core/picturesource.h:29]: Can you add a docstring that explains the behavior with respect to pictureFilePath and emitting the signal? With explicit recommendations for subclasses, something like "subclasses which implement their own file dialog should provide an empty string for \a pictureFilePath and emit \a pictureReceived when appropriate."

Comment (suricactus) [reply src/core/picturesource.h:37]: Done!

**Pull request merged**